core/app/models/spree/shipping_rate.rb

```diff
@@ -1,46 +1,56 @@
 module Spree
   class ShippingRate < Spree::Base
+    attr_writer :order
+
     belongs_to :shipment, class_name: 'Spree::Shipment'
     belongs_to :shipping_method, -> { with_deleted }, class_name: 'Spree::ShippingMethod', inverse_of: :shipping_rates
-    belongs_to :tax_rate, -> { with_deleted }, class_name: 'Spree::TaxRate'
 
-    delegate :order, :currency, to: :shipment
-    delegate :name, to: :shipping_method
+    has_many :taxes,
+             class_name: "Spree::ShippingRateTax",
+             foreign_key: "shipping_rate_id",
+             dependent: :destroy
+
+    delegate :currency, to: :shipment
+    delegate :order, to: :shipment, prefix: true
+    delegate :name, :tax_category, to: :shipping_method
     delegate :code, to: :shipping_method, prefix: true
+    alias_attribute :amount, :cost
 
-    def display_base_price
-      Spree::Money.new(cost, currency: currency)
-    end
+    extend DisplayMoney
+    money_methods :amount
+    alias_method :discounted_amount, :amount
 
-    def calculate_tax_amount
-      tax_rate.calculator.compute_shipping_rate(self)
+    def order
+      @order || shipment_order
     end
 
     def display_price
-      price = display_base_price.to_s
-      if tax_rate
-        tax_amount = calculate_tax_amount
-        if tax_amount != 0
-          if tax_rate.included_in_price?
-            if tax_amount > 0
-              amount = "#{display_tax_amount(tax_amount)} #{tax_rate.name}"
-              price += " (#{Spree.t(:incl)} #{amount})"
-            else
-              amount = "#{display_tax_amount(tax_amount * -1)} #{tax_rate.name}"
-              price += " (#{Spree.t(:excl)} #{amount})"
-            end
-          else
-            amount = "#{display_tax_amount(tax_amount)} #{tax_rate.name}"
-            price += " (+ #{amount})"
-          end
-        end
-      end
-      price
+      price = display_amount.to_s
+
+      return price if taxes.empty? || amount == 0
+
+      tax_explanations = taxes.map(&:label).join(tax_label_separator)
+
+      Spree.t :display_price_with_explanations,
+               scope: 'shipping_rate.display_price',
+               price: price,
+               explanations: tax_explanations
     end
     alias_method :display_cost, :display_price
 
-    def display_tax_amount(tax_amount)
-      Spree::Money.new(tax_amount, currency: currency)
+    def pre_tax_amount
+      default_vat_amount_sum = Spree::TaxRate.
```

Comment:
You're not getting mixed up, this is IMO a bug in the current taxation system. The `pre_tax_amount` is the basis on which to calculate VAT (another way to say this it's the actual value of the good). The current taxation system in Solidus allows that actual value to change under certain conditions ("the receiving country also has VAT"). If the receiving country does not have VAT, then we create this refund so it kinda sorta appears like we didn't charge it in the first place. 

This is actually a glimpse into Solidus' future taxation system: All the prices that come out of the backend (calculated prices as well as prices entered by admins) will include the VAT of ONE place. If we ship to another place with different VAT, then we have to deduct that VAT (only), and apply another one. 

I think I will extract this to its own PR as it's really quite a complex topic.